### ISSUE TITLE
IntelliSense using wrong std version with CMake Tools

### ISSUE THREAD

**Issue by rmccampbell** [microsoft/vscode-cmake-tools#2639]
### Brief Issue Summary

In a minimal CMake/C++ project with the Visual Studio 2022 generator, I have `set(CMAKE_CXX_STANDARD 20)` in my CMakeLists.txt, and it can build and run fine, but within VS Code IntelliSence seems to be stuck on C++14 and reports errors on any C++17+ features. Looking in the C/C++ debug logs I can see that for some reason the "custom configuration received" (-std:c++20) doesn't match the compilation args actually used (stdver: ms_c++14), see below.

If I disable "configurationProvider": "ms-vscode.cmake-tools" and instead set the cppStandard manually IntelliSense works just fine. Also if I instead set CMAKE_CXX_STANDARD to 17 it works fine yielding stdver: ms_c++17, and setting it to 23 yields stdver: ms_c++latest. I also tested the behavior is the same with Visual Studio 2019, so it doesn't seem to be specific to the compiler version.

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.68.1",
  "cmtVersion": "1.12.10",
  "configurations": [
    {
      "folder": "c:\\Users\\Ryan\\Documents\\coding\\cpp\\cmaketest2",
      "cmakeVersion": "3.23.0",
      "configured": true,
      "generator": "Visual Studio 17 2022",
      "usesPresets": false,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": true,
    "hasCodeModel": true,
    "activeBuildType": "Debug",
    "buildTypesSeen": [
      "Debug",
      "Release",
      "MinSizeRel",
      "RelWithDebInfo"
    ],
    "requests": [
      "file:///c%3A/Users/Ryan/Documents/coding/cpp/cmaketest2/main.cpp"
    ],
    "responses": [
      {
        "uri": "file:///c%3A/Users/Ryan/Documents/coding/cpp/cmaketest2/main.cpp",
        "configuration": {
          "includePath": [],
          "defines": [],
          "compilerPath": "c:/program files (x86)/microsoft visual studio/2022/buildtools/vc/tools/msvc/14.32.31326/bin/hostx64/x64/cl.exe",
          "compilerArgs": [],
          "compilerFragments": [
            "/DWIN32 /D_WINDOWS /W3 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1",
            "-std:c++20"
          ]
        }
      }
    ],
    "partialMatches": [],
    "targetCount": 3,
    "executablesCount": 1,
    "librariesCount": 0,
    "targets": [
      {
        "name": "ALL_BUILD",
        "type": "UTILITY"
      },
      {
        "name": "ZERO_CHECK",
        "type": "UTILITY"
      },
      {
        "name": "cmaketest2",
        "type": "EXECUTABLE"
      }
    ]
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": null
    }
  ]
}
```

### Debug Log

```shell
[main] Configuring folder: cmaketest2 
[main] Saving open files before configure/build
[driver] Start configure 
[driver] Running pre-configure checks and steps
[cmakefileapi-driver] Configuring using kit
[cmakefileapi-driver] Invoking CMake C:\Program Files\CMake\bin\cmake.EXE with arguments ["--no-warn-unused-cli","-DCMAKE_TOOLCHAIN_FILE:STRING=C:\\vcpkg\\scripts\\buildsystems\\vcpkg.cmake","-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE","-Sc:/Users/Ryan/Documents/coding/cpp/cmaketest2","-Bc:/Users/Ryan/Documents/coding/cpp/cmaketest2/build","-G","Visual Studio 17 2022"]
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_TOOLCHAIN_FILE:STRING=C:\vcpkg\scripts\buildsystems\vcpkg.cmake -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -Sc:/Users/Ryan/Documents/coding/cpp/cmaketest2 -Bc:/Users/Ryan/Documents/coding/cpp/cmaketest2/build -G "Visual Studio 17 2022"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19044.
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: C:/Users/Ryan/Documents/coding/cpp/cmaketest2/build
[cmakefileapi-parser] Read reply folder: c:\Users\Ryan\Documents\coding\cpp\cmaketest2\build\.cmake\api\v1\reply
[cmakefileapi-parser] Found index files: ["cache-v2-7209881dfb4aca772625.json","cmakeFiles-v1-a1a5a5645ea2adab66a3.json","codemodel-v2-dcb1229628f91cedfa10.json","directory-.-Debug-d0094a50bb2071803777.json","directory-.-MinSizeRel-d0094a50bb2071803777.json","directory-.-Release-d0094a50bb2071803777.json","directory-.-RelWithDebInfo-d0094a50bb2071803777.json","index-2022-07-07T03-48-22-0328.json","target-ALL_BUILD-Debug-42208e4299ca0bc33d79.json","target-ALL_BUILD-MinSizeRel-42208e4299ca0bc33d79.json","target-ALL_BUILD-Release-42208e4299ca0bc33d79.json","target-ALL_BUILD-RelWithDebInfo-42208e4299ca0bc33d79.json","target-cmaketest2-Debug-c30b88cc8fab1feb9a68.json","target-cmaketest2-MinSizeRel-12c5de2f4515ca4d87b6.json","target-cmaketest2-Release-8f94bf7987dedfd89f4b.json","target-cmaketest2-RelWithDebInfo-efa63cee41443cc94367.json","target-ZERO_CHECK-Debug-44528ac639ca007706e1.json","target-ZERO_CHECK-MinSizeRel-44528ac639ca007706e1.json","target-ZERO_CHECK-Release-44528ac639ca007706e1.json","target-ZERO_CHECK-RelWithDebInfo-44528ac639ca007706e1.json","toolchains-v1-e8e3bcf4478d44cb2afe.json"]
[cache] Reading CMake cache file c:/Users/Ryan/Documents/coding/cpp/cmaketest2/build/CMakeCache.txt
[cache] Parsing CMake cache string
[visual-studio] Patch Windows SDK bin path from C:\Program Files (x86)\Windows Kits\10\bin\x64 to C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64 for C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\Build\vcvarsall.bat
[visual-studio] OK running C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\Build\vcvarsall.bat x64, env vars: {...truncated...}
```

### Additional Information

Relevant snippet from C/C++ debug logs:
```
Checking for syntax errors: C:\Users\Ryan\Documents\coding\cpp\cmaketest2\main.cpp
cpptools/queryTranslationUnitSource: C:\Users\Ryan\Documents\coding\cpp\cmaketest2\main.cpp (id: 112)
Custom configurations received:
  uri: file:///c%3A/Users/Ryan/Documents/coding/cpp/cmaketest2/main.cpp
  config: {
  "includePath": [],
  "defines": [],
  "compilerPath": "c:/program files (x86)/microsoft visual studio/2022/buildtools/vc/tools/msvc/14.32.31326/bin/hostx64/x64/cl.exe",
  "compilerArgs": [],
  "compilerFragments": [
    "/DWIN32 /D_WINDOWS /W3 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1",
    "-std:c++20"
  ]
}
cpptools/didChangeCustomBrowseConfiguration
Attempting to get defaults from C++ compiler in "compilerPath" property: 'c:/program files (x86)/microsoft visual studio/2022/buildtools/vc/tools/msvc/14.32.31326/bin/hostx64/x64/cl.exe'
cpptools/didChangeCustomConfiguration
Attempting to get defaults from C++ compiler in "compilerPath" property: 'c:/program files (x86)/microsoft visual studio/2022/buildtools/vc/tools/msvc/14.32.31326/bin/hostx64/x64/cl.exe'
cpptools/resumeParsing
sending compilation args for C:\Users\Ryan\Documents\coding\cpp\cmaketest2\main.cpp
  include: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2022\BUILDTOOLS\VC\TOOLS\MSVC\14.32.31326\INCLUDE
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UM
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UCRT
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\SHARED
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\WINRT
  include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\CPPWINRT
  define: _DEBUG
  define: _MT
  define: _DLL
  define: WIN32
  define: _WINDOWS
  other: --rtti
  stdver: ms_c++14
  intelliSenseMode: windows-msvc-x64
```

**Comment by bobbrow** [comment]
This looks like a duplicate of #2608.  The issue is in the C++ extension.  There is a pre-release available with the fix (1.11.0). Could you give that a try and let us know if you still have an issue?

**Comment by bobbrow** [comment]
Thanks for confirming.  1.10.8 (stable) was also released yesterday and has the fix.
